subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
```

Comment:
I may imagine you're commented on the wrong line. Anyway, two points: 1) This code sets the local address, which is bound to an interface, so if it sets address, it also sets interface. 2) I spent reasonable time to look into code, and on "outgoing" path, only bind() sets local address and iface (all other paths appear to be for incoming packets/connections). 3) Final point obvious from code is that iface is not set if address is already set, because p.1 (they always set together).